packages/ra-core/src/form/useInput.ts

```diff
@@ -49,7 +49,7 @@ export const useInput = <ValueType = any>(
     const defaultId = useId();
 
     if (
-        !source &&
+        !finalSource &&
         props.label == null &&
         process.env.NODE_ENV === 'development'
     ) {
```

Comment:
I agree we should change this condition, thanks for your report!

Actually I wonder if we should go even further. AFAICT this check was mainly added to make sure we won't call [useController](https://react-hook-form.com/docs/usecontroller) with an empty `name` (and also, to make TS happy).

So we should maybe replace it altogether by:

```js
    if (
        !finalName &&
        process.env.NODE_ENV === 'development'
    ) { /* ... */ }
```

@fzaninotto wdyt?

Reply:
You're right, an input with no source but with a name should not raise any warning!

Reply:
@slax57 @fzaninotto thanks for your feedback! Are you guys going to work on this? I saw the changes las week in [useInput](https://github.com/marmelab/react-admin/blame/master/packages/ra-core/src/form/useInput.ts) but I'm still getting the warnings which is making my application very slow when editing/creating
Thanks!

Reply:
@GuilhermeCarra the changes you refer to are unrelated (you can have a look at the originating [PR](https://github.com/marmelab/react-admin/pull/10148) description to understand the context). 

Regarding this PR, if you apply the changes I requested [above](https://github.com/marmelab/react-admin/pull/10153#discussion_r1723486408) we will be happy to merge it. Thanks!

Reply:
@slax57 thanks! I've made the change 🙌